src/include/packager/repositories_include.rb

```diff
@@ -464,7 +464,7 @@ def StoreSource
 
       if ret == :again
         return :back
-      elsif ret == :abort
+      elsif ret == :abort || ret == :cancel
         return :abort
       end
       :next
```

Comment:
OK, makes sense, originally I wanted as minimal change as possible, but this looks really better...